runtime/common/src/precompile/homa.rs

```diff
@@ -0,0 +1,442 @@
+				);
+
+				<module_homa::Pallet<Runtime> as HomaManager<Runtime::AccountId, Balance>>::request_redeem(
+					who, amount, fast_match,
+				)
+				.map_err(|e| PrecompileFailure::Revert {
+					exit_status: ExitRevert::Reverted,
+					output: Into::<&str>::into(e).as_bytes().to_vec(),
+					cost: target_gas_limit(target_gas).unwrap_or_default(),
+				})?;
+
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: vec![],
+					logs: Default::default(),
+				})
+			}
+			Action::GetExchangeRate => {
+				let rate =
+					<module_homa::Pallet<Runtime> as HomaManager<Runtime::AccountId, Balance>>::get_exchange_rate();
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128(rate.into_inner()),
+					logs: Default::default(),
+				})
+			}
+			Action::GetEstimatedRewardRate => {
+				let rate = <module_homa::Pallet<Runtime> as HomaManager<Runtime::AccountId, Balance>>::get_estimated_reward_rate();
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128(rate.into_inner()),
+					logs: Default::default(),
+				})
+			}
+			Action::GetCommissionRate => {
+				let rate =
+					<module_homa::Pallet<Runtime> as HomaManager<Runtime::AccountId, Balance>>::get_commission_rate();
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128(rate.into_inner()),
+					logs: Default::default(),
+				})
+			}
+			Action::GetFastMatchFee => {
+				let rate =
+					<module_homa::Pallet<Runtime> as HomaManager<Runtime::AccountId, Balance>>::get_fast_match_fee();
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Returned,
+					cost: gas_cost,
+					output: Output::default().encode_u128(rate.into_inner()),
+					logs: Default::default(),
+				})
+			}
+		}
+	}
+}
+
+struct Pricer<R>(PhantomData<R>);
+
+impl<Runtime> Pricer<Runtime>
+where
+	Runtime: module_evm::Config + module_homa::Config + module_prices::Config,
+{
+	const BASE_COST: u64 = 200;
+
+	fn cost(
+		input: &Input<Action, Runtime::AccountId, Runtime::AddressMapping, Runtime::Erc20InfoMapping>,
+	) -> Result<u64, PrecompileFailure> {
+		let action = input.action()?;
+
+		let cost: u64 = match action {
+			Action::Mint => {
+				let cost = InputPricer::<Runtime>::read_accounts(1);
+				let weight = <Runtime as module_homa::Config>::WeightInfo::mint();
+
+				cost.saturating_add(WeightToGas::convert(weight))
+			}
+			Action::RequestRedeem => {
+				let cost = InputPricer::<Runtime>::read_accounts(1);
+				let weight = <Runtime as module_homa::Config>::WeightInfo::request_redeem();
+
+				cost.saturating_add(WeightToGas::convert(weight))
+			}
+			Action::GetExchangeRate => {
+				// Homa::TotalVoidLiquid (r: 1)
+				// Homa::ToBondPool (r: 1)
+				// Tokens::TotalIssuance(r: 1)
+				// Homa::StakingLedgers(r: ?)
```

Comment:
is this verified to be 1?

Reply:
Yeah that's the homa benchmarking issue #2084. 

It depends on storage for number of reads (three reads on karura right now), needs to be fixed as currently `mint` and `request_redeem` extrinsic are wrong in the same manner